src/software/thunderscope/robot_diagnostics/robot_info.py

```diff
@@ -387,3 +407,50 @@ def __calculate_average_round_trip_time(self, new_time: float) -> int:
         self.previous_primitive_rtt_values.append(new_time)
         sum_rtt_time_milliseconds = sum(self.previous_primitive_rtt_values)
         return int(sum_rtt_time_milliseconds / len(self.previous_primitive_rtt_values))
+
+    def __run_ansible_playbook(self, playbook: str) -> None:
+        """Runs an Ansible playbook on this robot.
+
+        This method runs run_ansible.py via tbots.py to trigger a bazel build
+        of Thunderloop. Certain playbooks (e.g. flashing Thunderloop) require that
+        robot software binaries are built ahead of the playbook execution.
+
+        Ansible connects to the robot via SSH. If there are no previously saved SSH
+        credentials or if connecting to the robot previously failed, the user will be
+        prompted to enter the robot's IP/hostname and user password.
+
+        :param playbook: the playbook to run
+        """
+        if not self.ssh_connected:
+            dialog_accepted = self.ssh_dialog.exec()
+            if not dialog_accepted:
+                return
+
+        def restart_thunderloop() -> AnsibleResult:
```

Comment:
nit: method doesn't only deal with restarting thunderloop from what I can see, so maybe `run_playbook_helper` or something idk

same for the `finished` method